zyte_spider_templates/spiders/ecommerce.py

```diff
@@ -266,3 +288,17 @@ def get_parse_product_request(
         )
         scrapy_request.meta["allow_offsite"] = True
         return scrapy_request
+
+    def page_params_for_heuristics(
+        self, response: DummyResponse
```

Comment:
what do you think about passing page_params here? Or maybe response.meta. It looks weird that we require DummyResponse; in most cases parameters annotated as DummyResponse shouldn't be used in the body.

Reply:
That's much better! https://github.com/zytedata/zyte-spider-templates/pull/52/commits/eb98f4e0bfbe02c2dfa72b001b4d400378a6c914